src/variable.c

```diff
@@ -521,11 +521,15 @@ mrb_obj_iv_inspect(mrb_state *mrb, struct RObject *obj)
 MRB_API mrb_value
 mrb_iv_remove(mrb_state *mrb, mrb_value obj, mrb_sym sym)
 {
-  mrb_check_frozen(mrb, mrb_obj_ptr(obj));
-  if (obj_iv_p(obj)) {
+  if (mrb_immediate_p(obj)) {
+    mrb_raisef(mrb, E_FROZEN_ERROR, "can't modify frozen %S",
+               mrb_obj_value(mrb_class(mrb, obj)));
+  } else if (obj_iv_p(obj)) {
```

Comment:
I'm the one who made the related change. If you call `mrb_raisef()` directly, the binary size will increase, so I think you should use `mrb_frozen_error()`.

example: https://github.com/shuujii/mruby/commit/f422d1434b

The signature of `mrb_frozen_error()` has been changed because it was originally inappropriate.

Reply:
Your patch looks better, with the changes to the signature of `mrb_frozen_error`. What about the tests, though? I think I'll incorporate your changes into my PR.

Reply:
Your tests looks good to me (however, `.freeze` may not be necessary).

Reply:
Oops, yes, that was left there by accident from something I had earlier.